### ISSUE TITLE
DON'T UPDATE YOUR STATIONS!

### ISSUE THREAD

**Issue by Patrick762** [warhammerkid/bluetti_mqtt#101]
The current update for at least the EP600 renames the Bluetooth name to "PBOX".

This causes some problems with this software.

Thanks @KM011092 for this info in your Pull request.

@warhammerkid could you check if it is possible to differentiate the stations by their **device_type** field?
If I find some spare time I'll also look into this because it also causes my HA integration to no longer work for new devices.

**Comment by twister36** [comment]
Interesting... I wonder if this is related to an issue I'm looking into for a friend. They updated their AC500 firmware on one unit and now they can't connect to their unit via bluetti-mqtt. 

Did you ever resolve this?

**Comment by twister36** [comment]
> The current update for at least the EP600 renames the Bluetooth name to "PBOX".
> 
> This causes some problems with this software.
> 
> Thanks @KM011092 for this info in your Pull request.
> 
> @warhammerkid could you check if it is possible to differentiate the stations by their **device_type** field? If I find some spare time I'll also look into this because it also causes my HA integration to no longer work for new devices.

Can you provide your steps to verify the bluetooth name?

**Comment by Patrick762** [comment]
You can check it if you open your Bluetooth settings on your phone

**Comment by Patrick762** [comment]
> Interesting... I wonder if this is related to an issue I'm looking into for a friend. They updated their AC500 firmware on one unit and now they can't connect to their unit via bluetti-mqtt. 
> 
> Did you ever resolve this?

I don't know how to resolve it in this repository, but I have a workaround in my HA integration (selecting the model manually). Next step would be to automate it again

**Comment by twister36** [comment]
> > Interesting... I wonder if this is related to an issue I'm looking into for a friend. They updated their AC500 firmware on one unit and now they can't connect to their unit via bluetti-mqtt.
> > Did you ever resolve this?
> 
> I don't know how to resolve it in this repository, but I have a workaround in my HA integration (selecting the model manually). Next step would be to automate it again

Which HA integration are you using? I'm just using the native bluetti-mqtt python app to send the data to MQTT. Is there a working integration now?

**Comment by Patrick762** [comment]
My integration is available in the HACS Store as "Bluetti BT". It only has controls for AC and DC outputs but I plan on getting some other Bluetti stations to test the other controls.

**Comment by Patrick762** [comment]
To use it with the PBOX update you'll need the latest Beta update

**Comment by twister36** [comment]
> My integration is available in the HACS Store as "Bluetti BT". It only has controls for AC and DC outputs but I plan on getting some other Bluetti stations to test the other controls.

Ok, let me take a look. I was starting to write my own and got as far as connecting to the Bluetti device and started to to discover the Services and Chrematistics of the device.... but Ideally I didn't want to reinvent the wheel. It appears that yours is still based of this repository (bluetti-mqtt).  correct?

**Comment by twister36** [comment]
> To use it with the PBOX update you'll need the latest Beta update

Can you clarify what you are referring to ? I assume you mean the HACS needs to be the beta?

**Comment by Patrick762** [comment]
> > My integration is available in the HACS Store as "Bluetti BT". It only has controls for AC and DC outputs but I plan on getting some other Bluetti stations to test the other controls.
> 
> Ok, let me take a look. I was starting to write my own and got as far as connecting to the Bluetti device and started to to discover the Services and Chrematistics of the device.... but Ideally I didn't want to reinvent the wheel. It appears that yours is still based of this repository (bluetti-mqtt).  correct? 

Yes. Currently it is based on this repository. But with future HA updates, one of the dependencies used will no longer work, so I'll have to switch to an own library.

**Comment by Patrick762** [comment]
> > To use it with the PBOX update you'll need the latest Beta update
> 
> Can you clarify what you are referring to ? I assume you mean the HACS needs to be the beta?

You need the beta version 0.0.15 or 0.0.16 of the bluetti_bt integration (switch on when installing). You don't need HACS beta.

**Comment by twister36** [comment]
> > > To use it with the PBOX update you'll need the latest Beta update
> > 
> > 
> > Can you clarify what you are referring to ? I assume you mean the HACS needs to be the beta?
> 
> You need the beta version 0.0.15 or 0.0.16 of the bluetti_bt integration (switch on when installing). You don't need HACS beta.

Ok, I just added it and it did auto discovery all 6 bluetti devices... which one of them is the bluetti with the issue. However, after "configuring" the integration for that device all entities are 'unknown'. Instead of asking questions about your project here, I'll create an issue on your project so we can work together, as I have a total of 12 bluetti devices that we can test with.

**Comment by tschernet** [comment]
Hi, I´ve been using bluetti_mqtt on a raspberry pi to get data from my EP600 and send it to my local mqtt broker. Unfortunately, I did update my bluetti´s iot thingie last week (yeah, stupid, I know... ), and now bluetti_mqtt --scan does not find any device.

According to pip, I have bluetti_mqtt version 0.15.0 installed. Can anybody direct me to what I need to do to get comms back? Is there some way to apply the above mentioned workaround to bluetti_mqtt? Thanks!

**Comment by Patrick762** [comment]
The biggest problem is, they name multiple powerstations the same way, so this software can't detect the type anymore. If you're using Home Assistant, you could use my integration, otherwise you can try to add the type detection to this software like I have it [here](https://github.com/Patrick762/hassio-bluetti-bt/blob/dev/custom_components/bluetti_bt/bluetti_bt_lib/bluetooth/device_recognizer.py)

**Comment by tschernet** [comment]
>The biggest problem is, they name multiple powerstations the same way

I see. I´d be happy to set the device type manually (e.g. by some commandline parameter)

> If you're using Home Assistant, you could use my integration,

Unfortunately I´m not; I am using NodeRed to process the data and show a dashboard, plus log into influxdb and use grafana for plotting

> otherwise you can try to add the type detection to this software like I have it here

Thanks a lot, I´ll look into it. I´m not much of a python guy myself, but I´ll see if I can find my way around. (I´ll be grateful for any hints though!)

**Comment by KM011092** [comment]
Is this a new issue or related to the rename of the EP600 to PBOX ? If itˋs caused by the rename to PBOX you may want to check the fork I am running on a RaspberryPi to monitor my EP600: https://github.com/KM011092/bluetti_mqtt.git. If this is a new issue, I better not update the IOT :-) and would be very interested if you found a way to fix it.

**Comment by tschernet** [comment]
> Is this a new issue or related to the rename of the EP600 to PBOX ? If itˋs caused by the rename to PBOX you may want to check the fork I am running on a RaspberryPi to monitor my EP600: https://github.com/KM011092/bluetti_mqtt.git. 

Hi, I´d love to do that! Please ignore my ignorance, but how do I tell pip to install your fork instead of the original project? Both mention the same command (pip install bluetti_mqtt), so I don´t know how to switch?

Thanks a lot! brgds - tscher


If this is a new issue, I better not update the IOT :-) and would be very interested if you found a way to fix it.

**Comment by KM011092** [comment]
Hi, try pip install git+ https://github.com/KM011092/bluetti_mqtt.git

You may want to uninstall your old version first. 
Please be aware that I am not a developer. The version is working for me, but I will not be able to give a lot of support if it does not work in your scenario.

**Comment by tschernet** [comment]
ok, did that; unfortunately, when running --scan, it still does not detect my EP600...

**Comment by KM011092** [comment]
Mhhh, then this seem to be a new issue and I better not update my EP600 as long as everything is working for me. Sorry that this did not help.

**Comment by tschernet** [comment]
> Mhhh, then this seem to be a new issue and I better not update my EP600 as long as everything is working for me. Sorry that this did not help.

No worries, thanks for your and everybody´s great work in creating all this in the first place, and for all the support you are providing!
